src/settings.js

```diff
@@ -62,6 +62,15 @@ var TOTAL_STACK = 5*1024*1024; // The total stack size. There is no way to enlar
 var TOTAL_MEMORY = 16777216;     // The total amount of memory to use. Using more memory than this will
                                  // cause us to expand the heap, which can be costly with typed arrays:
                                  // we need to copy the old heap into a new one in that case.
+var MALLOC = "dlmalloc"; // What malloc()/free() to use, out of
+                         //  * dlmalloc - a powerful general-purpose malloc
+                         //  * emmalloc - a simple and compact malloc designed for emscripten
+                         // dlmalloc is necessary for multithreading, split memory, and other special
+                         // modes, and will be used automatically in those cases.
+                         // In general, if you don't need one of those special modes, and if you don't
+                         // allocate very many small objects, you should use emmalloc since it's
+                         // smaller. Otherwise, if you do allocate many small objects, dlmalloc
+                         // is usually worth the extra size.
```

Comment:
I wonder if an option `"none"` will be useful, to force user to supply implementation of `malloc` and `free` themselves? Not to say that should be implemented in this PR, but just wondering if that'll be feasible in this arch?

Reply:
I think regardless of what this setting is, the linking of malloc should behave as it does on native platforms; i.e. if you put your own implementation of `malloc` and `free` in an object file in the link, then that will be used instead of the emscripten-provided one.

Reply:
Yeah, that's right, and that's something we do already test to work. This replacement however can be a bit of magic, since if something changes, e.g. one forgets to link to own malloc, there is generally no error message, and one silently gets the built-in malloc, so thought this kind of option would be nice to explicitly kill the original malloc from being considered, in which case one would get a nice missing symbol error instead if e.g. after build system refactoring one lacks linking in their own malloc.

Reply:
Yeah, we could add a "none" option that means it would be a link error to not provide your own malloc. I wonder if there isn't a more general way to do this. Not opposed, though.